src/services/balancer/contracts/contracts/vault.ts

```diff
@@ -90,22 +90,30 @@ export default class Vault {
     type: PoolType,
     tokens: TokenInfoMap
   ) {
-    if (type == 'Weighted' || type === 'Investment') {
+    if (this.isWeightedLikePool(type)) {
       const totalWeight = weights.reduce((a, b) => a.add(b), BigNumber.from(0));
       return weights.map(
         w =>
           ((100 / Number(formatUnits(totalWeight, 10))) *
             Number(formatUnits(w, 10))) /
           100
       );
-    } else if (type === 'Stable' || type === 'MetaStable') {
+    } else if (this.isStableLikePool(type)) {
       const tokensList = Object.values(tokens);
       return tokensList.map(() => 1 / tokensList.length);
     } else {
       return [];
     }
   }
 
+  public isWeightedLikePool(type: PoolType) {
+    return ['Weighted', 'Investment', 'LiquidityBootstrapping'].includes(type);
+  }
+
+  public isStableLikePool(type: PoolType) {
+    return ['Stable', 'MetaStable'].includes(type);
+  }
+
```

Comment:
Is there a nice way we can support these type checks in [usePool](https://github.com/balancer-labs/frontend-v2/blob/develop/src/composables/usePool.ts)? I know we're currently only accepting the pool as input there, maybe we can change that? Then change all references to it to pass in only the poolType.

Reply:
Yep! fixed in https://github.com/balancer-labs/frontend-v2/pull/822/commits/cd43eb6a84b61259b5e1776a609d2f4da62fcb51